src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -365,7 +365,7 @@ private ReturnTypeChain BuildReturnTypes()
                     {
                         if (convenienceParameter.Type is { IsFrameworkType: false, Implementation: ModelTypeProvider model })
                         {
-                            var parameters = BuildSpreadParameters(model).OrderBy(p => p.DefaultValue == null ? 0 : 1);
+                            var parameters = BuildSpreadParameters(model).OrderBy(p => p.DefaultValue == null ? 0 : 1).ToArray();
```

Comment:
Why do we add `ToArray()`?

Reply:
I changed the signature of another method which consumes parameters from `IEnumerable<Parameter>` to `IReadOnlyList<Parameter>`.
This does not impact the performance really because we will call `ToArray` any way later